entitlement/src/main/java/org/killbill/billing/entitlement/engine/core/DefaultEventsStream.java

```diff
@@ -516,7 +516,8 @@ private void computeEntitlementBlockingStates() {
 
     private List<BlockingState> filterBlockingStatesForEntitlementService(final BlockingStateType blockingStateType,
                                                                           @Nullable final UUID blockableId) {
-    	return blockingStates.stream()
+    	final Collection<BlockingState> blockingStatesToConsider = includeDeletedEvents ? blockingStatesWithDeletedEvents : blockingStates;
```

Comment:
Hum... The issue may be that this is called by `DefaultEventsStream#computeEntitlementBlockingStates` to compute `subscriptionEntitlementStates` in the `setup` method. So this would mean that we would alway return the `deleted` blocking states for `ENTITLEMENT_SERVICE` is the instance variable was set with `includeDeletedEvents`.

I think we want to never return any deleted blocking states (or subscription events) unless explicitly called from the `DefaultSubscription#getSubscriptionEvents`

Reply:
Yes, I'm not happy with this change either, I had to make this change to prevent [this](https://github.com/reshmabidikar/killbill/blob/31f26ddc7522f440b7f4c19327f6406ec7424af5/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestSubscription.java#L1014) addon test from failing as mentioned [here](https://github.com/killbill/killbill/pull/1762#issuecomment-1230094693). 

So basically, without this change, the [entitlementCancelEvent](https://github.com/reshmabidikar/killbill/blob/835a6a5cf9344da4effd058bd05167f6b6301794/entitlement/src/main/java/org/killbill/billing/entitlement/engine/core/DefaultEventsStream.java#L81) field does not get populated due to which the `STOP_ENTITELMENT` event is not returned for the addon subscription causing the test to fail [here](https://github.com/reshmabidikar/killbill/blob/31f26ddc7522f440b7f4c19327f6406ec7424af5/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestSubscription.java#L1118).


The alternative would be maybe to create a separate list like `subscriptionEntitlementStatesWithDeletedEvents` and use that to compute `entitlementCancelEvent` and other fields as required. Please let me know what you think.

Reply:
Reverted these changes and updated test.